### PR TITLE
device_token accept binary and list；fix some elvis errors

### PR DESCRIPTION
**Pull request by zhongwencool**
1. Device_token accept binary 
   Sometimes we store device_token in binary, we don't want to transform binary to list every time
  
   Another reason is binary is faster than list, see benchmark test on 
   https://gist.github.com/zhongwencool/659442f5e51a22c97ae1
```erlang
  1> hex:test().
  true
  BinTime:1927792 StrTime:2175268 Diff:-247476
  ok
```
       Maybe  hexstr_to_bin/1 could accept binary will be better?

2. fix some elvis errors 

3. update elvis.config

### CODE REVIEW

File: elvis.config [hunk 1]
```diff
@@ -3,42 +3,41 @@
    elvis,
    [
     {config,
-     [#{dirs => ["src", "test"],
+     [#{dirs => ["src"],
```
Comment (elbrujohalcon) [inline elvis.config]: If you're about to remove the test folder from here, please add another config group to check it.

File: elvis.config [hunk 2]
```diff
@@ -3,42 +3,41 @@
...
         filter => "*.erl",
-        rules => [{elvis_style, line_length, #{limit => 80}},
```
Comment (elbrujohalcon) [inline elvis.config]: Why are you going back to an older format for elvis.config? (the one with maps is the current canonical one)

Comment (zhongwencool) [inline elvis.config]: I just git clone apns run 
```erlang
make elvis
```
it just update elvis.config all by this 

Maybe I should revert elvis.config?

Comment (elbrujohalcon) [inline elvis.config]: Yeah… newer versions of erlang.mk use newer versions of elvis, like gadget does.
The best thing to do will probably be to keep elvis.config as it was and eventually someone at @inaka will upgrade erlang.mk… and then we can start running make elvis again.
Sorry for the inconvenience, @zhongwencool

**Comment by zhongwencool** [comment]
@elbrujohalcon  already revert elvis

Thanks for this great project.  :)

:+1:

**Comment by elbrujohalcon** [comment]
thank you, @zhongwencool :)

**Pull request merged**